### ISSUE TITLE
Game crashes when editor is moved to second monitor [Win10][Dual Monitors]

### ISSUE THREAD

**Issue by Laroy-Milton** [godotengine/godot#28785]
**Godot version:**
3.1.1

**OS/device including version:**

Windows 10
GeForce GTX 1080/PCIe/SSE2

**Issue description:**

The editor opens on screen 1. When I play the game in the same screen there is no issue. When I drag the editor to screen 2 then hit play again, the game crashes instantly.
When I play from screen 1 then drag the game window to screen two the game continues to work. Both monitors are at the same resolution but are not the same model. 

drivers/unix/net_socket_posix.cpp:190 - Socket error: 10054
Is displayed after killing the process.

**Comment by 2shady4u** [comment]
Hello!
I can't seem to replicate this error on my machine unfortunately...

**Godot version:**
3.1.1

**OS/device including version:**
Windows 10
GeForce GTX 1050 Ti with Max-Q Design

Proof of no-crash:
https://vimeo.com/335461854

Is my attempt at crash replication correct?
What should I do to replicate your crash?

**Comment by CraftyManiac** [comment]
Yes that's exactly right! My monitors are connected to seperate ports on my GPU. One is HDMI, the other is dvi

**Comment by kirkfriend** [comment]
I am experiencing the same issue here. I have both monitors running of a RX 580 via HDMI. This happens in both GLES2 and GLES3 using Godot 3.1.1 from Steam.

If I change my primary monitor it will reverse the problem.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by damymetzke** [comment]
I have the same issue on Godot **3.2.3**.
Steps are identical to those described by the author.
However I do not get the error the autor describes:

> drivers/unix/net_socket_posix.cpp:190 - Socket error: 10054

I'm using the prebuild stable *mono* release.

|                      |                                          |
|---------------|-----------------------------|
|Graphics card|NVIDIA GeForce RTX 2060|
|OS                 |Windows 10                       |

I'm using a laptop and my second monitor is connected through HDMI.

**Comment by damymetzke** [comment]
I've done some testing and figured out why it was not working on my end.
In short it is an issue with an external DLL from Nahimic.
Disabling the sound tracker engine, which is responsible for injecting the DLL, will solve the issue.

To my knowledge the following conditions must be met in order for the bug to occur:
- The game must be running GLES2 or GLES3
- The computer must have Nahimic installed, and the 'sound tracker engine' enabled
- The game must be initially opened on a non-primary monitor

*When running Vulkan I have observed the game window initially being opened on the primary monitor,
and then being moved to the secondary monitor.
This could explain why Vulkan works, but I cannot veriyfy.*

What I have observed:
- I tested the master branch, using Vulkan, which resulted in no issues.
*I did not test GLES2 on the master branch*
- I tested Both GLES3 and GLES2 on the 3.2 branch, both resulted in the issue.
- Using the visual studio debugger(on the 3.2 branch) I have concluded that the program gets stuck `platform/windows/context_gl_windows.cpp`
It gets stuck at ContextGL_Windows::swap_buffers.
The first statement `SwapBuffers(hDC)` is called, execution never reaches the next line.
- According the the call stack the game is stuck within the function `RasterizerGLES3::set_boot_image`,
- After this the call stack continued in external code with no pdp available
- Included in the external code was `NahimicOSD.dll`

Nahimic is a program that is bundled with, to my knowledge, every MSI laptop.
It is an audio program, however it has a feature called 'sound tracker'.
This converts 3D audio into a direction, *and can display this using an overlay*.
Presumable it injects its DLL during boot.

After this I went into the Nahimic app, where I disabled the 'sound tracker engine'.
When I disabled this the issue was resolved.

The best theory I can come up with is that is some interaction some of all of these:
- The nahimic DLL
- GLES
- A non primary monitor
- The boot image

I do not have the skills to figure out any more than this.

I'll be sure to reply if any more information is required.

**Comment by KoBeWi** [comment]
@CraftyManiac Would be useful to know if your issue is the same as comment above or similar.

I assume that other applications don't crash because of Nahimic, so there's some bug on Godot side to fix.

**Comment by damymetzke** [comment]
I've done a little more testing and can confirm the following:
- The program always gets stuck at the first time the buffers are swapped,
removing the boot image only postpones the issue to the end of the first draw call.
- The issue is triggered when the majority of the window is on the secondary monitor.
Forcefully creating the window on the second monitor still triggers the issue.
Creating the window on the secondary monitor,
but moving it to the primary monitor using the `--position` flag,
will **not** trigger the issue.

I've tried running the game with bot Steam and Discord overlays,
neither could trigger the issue.
Both overlays were only available a few seconds after startup,
which could be the reason the issue did not trigger.
Perhaps they have been designed this way for this specific reason,
but there is no concrete evidence to support this.

**Comment by RufusWein** [comment]
I has same problema.

Execution on other monitor (secondary) show "No respond" in Godot Windows. I did testing for GLES 2 and GLES3 and is same, doesnt work.

My configuration is:
Godot version: ![image](https://user-images.githubusercontent.com/10912659/108057450-f1207f00-7052-11eb-9d01-6ea666e33d53.png)
SO: Windows 10
GPU: RTX 2060
Memory: lot!!

**Comment by RufusWein** [comment]
Ok, good news

### This is a Windows 10 bug, update Windows 10 and it is will fix.

**Comment by Supergeek** [comment]
> 
> 
> Ok, good news
> ### This is a Windows 10 bug, update Windows 10 and it is will fix.

Which Windows 10 update fixes this issue? I just ran into this bug today, seven months and a point release later. First time trying to run a Godot Scene on my secondary monitor. 

* Godot 3.3.3 with Mono 
* Windows 10 version 2004 (build 19041.1165) 
* Nahimic sound drivers (desktop PC, AMD 3900X, GRX 3060, not a laptop) 
* Two displays, both connected via DisplayPort


Googling gave me this nugget:

> workaround is to put OS.center_window() in one of your node's _ready() functions

I tried that and it did not work.

**Comment by igorgsouza** [comment]
Having the same problem here, did anyone manage to find a fix?

I'm using Godot 3.4

**Comment by Trikisatan** [comment]
Still a problem in Godot 4.

**Comment by Calinou** [comment]
- Duplicate of https://github.com/godotengine/godot/issues/13208 (same cause).

**Comment by ronindesign** [comment]
Still an issue on Godot (v4.3.beta1.official [a4f2ea91a],) Windows 11, Nvidia GeForce RTX 4090.

After selecting a project and moving editor window to full screen in my 2nd or 3rd monitor, Godot immediately freezes.

**Comment by Kearss** [comment]
Just a nugget for someone with similar issues.
I had the same problem. 
Tried a fresh install of windows 11 as well and that didn't work.
I don't think it has anything to do with graphics since i tried 2-3 different graphic drivers from Amd but nothing.
The issue is with nahimic services. I don't know any specifics why. 
But disabling it from system configuration godot has been working fine on my second monitor. Also tried turning it back on and the same issue occurred.
 
in conclusion 
disabling the nahimic service might solve your problem for now. 
I don't know if this fix will work for win10 but I don't see why not.
Win11 it did work for me, although my win version is 24H2
one monitor is 1080p and another is 1440p
Godot might want to look into the nahimic service.

**Comment by AThousandShips** [comment]
Please see the issue this is a duplicate of, this has been solved in 4.4, please do not add any more details on this as it is a duplicate, if you have any details (after testing with 4.4) please add them to:
* #13208
